Some package updates

nanoant opened the pull request:
I have managed to fix some small package problems, update few existing, and introduce couple of interesting ones :)
LLVM and Clang updates are still on my list, also I am planning to introduce qemu.

nanoant committed:
Update to pcre 8.43

Ported from using Arch

nanoant committed:
Update to the_silver_searcher 2.2.0

Ported from Arch

nanoant committed:
Add glib2 2.56.4

Partially ported from Arch and Homebrew

This is the last version that comes with configure and does not need meson.

nanoant committed:
Add slang 2.3.2

Ported from Arch

nanoant committed:
Add mc 4.8.22

Ported from Arch and Homebrew

nanoant committed:
Update to gettext 0.20.1

Including patch for macOS warning verbosity from:
https://git.merproject.org/mirror/gettext/commit/ec0e6b307456ceab352669ae6bccca9702108753.diff

Read more at: https://github.com/Homebrew/homebrew-core/issues/41139

nanoant committed:
Fix pacman package

1. Dependency package fakechroot does not exist, change to fakeroot,
2. Add openssl to dependencies and point to /opt/arch LIBSSL

nanoant committed:
Update to fakeroot 1.24

nanoant committed:
Update to python3 3.7.4

nanoant committed:
Add ninja 1.9.0

Ported from Arch

nanoant committed:
Add re2c 1.2.1

Ported from Arch

nanoant committed:
Fix pacman touch epoch

macOS/BSD touch does not know how to parse epoch time.

nanoant committed:
Update python3 libraries

Ported from Arch

python3-appdirs 1.4.3
python3-packaging 19.2
python3-pyparsing 2.4.5
python3-setuptools 41.4.0
python3-six 1.13.0

nanoant committed:
Add meson 0.52.0

Ported from Arch

nanoant committed:
Fix docbook-xsl catalog

It should point to /opt/arch/share/xml/ not /usr/share/xml/.

nanoant committed:
Update to glib2 2.62.3

This one now is built with meson.

lloeki commented:
Super cool. I think I have `qemu` locally somewhere but it's hackishly packaged.

Also, a quick note: `.install` scripts that run info tools seem to not work correctly and output failures on install (GNU stuff, e.g `bash`), I was set on just dropping the `info` part of those.

nanoant committed:
Add python-sphinx 2.2.1 and its dependencies

Packages mainly ported from Arch Linux

Dependencies for Sphinx added/updated:

python3-3.7.4-1-x86_64
python3-apipkg-1.5-1-any
python3-appdirs-1.4.3-1-any
python3-attrs-19.3.0-1-any
python3-babel-2.7.0-1-any
python3-certifi-2019.9.11-1-any
python3-chardet-3.0.4-1-any
python3-docutils-0.14-1-any
python3-html5lib-1.0.1-1-any
python3-idna-2.8-1-any
python3-imagesize-1.1.0-1-any
python3-importlib-metadata-0.23-1-any
python3-iniconfig-1.0.0-1-any
python3-jinja-2.10.3-1-any
python3-markupsafe-1.1.1-1-x86_64
python3-more-itertools-7.2.0-1-any
python3-packaging-19.2-1-any
python3-pluggy-0.13.1-1-any
python3-py-1.8.0-1-any
python3-pygments-2.4.2-1-any
python3-pyparsing-2.4.5-1-any
python3-pytest-5.3.0-1-any
python3-pytest-runner-5.2-1-any
python3-pytz-2019.3-1-any
python3-requests-2.22.0-1-any
python3-setuptools-41.4.0-1-any
python3-setuptools-scm-3.3.3-1-any
python3-six-1.13.0-1-any
python3-snowballstemmer-2.0.0-1-any
python3-sphinx-2.2.1-1-any
python3-sphinx-alabaster-theme-0.7.12-1-any
python3-sphinxcontrib-applehelp-1.0.1-1-any
python3-sphinxcontrib-devhelp-1.0.1-1-any
python3-sphinxcontrib-htmlhelp-1.0.2-1-any
python3-sphinxcontrib-jsmath-1.0.1-1-any
python3-sphinxcontrib-qthelp-1.0.2-1-any
python3-sphinxcontrib-serializinghtml-1.1.3-1-any
python3-urllib3-1.25.7-1-any
python3-wcwidth-0.1.7-1-any
python3-webencodings-0.5.1-1-any
python3-zipp-0.6.0-1-x86_64

nanoant committed:
Update to cmake 3.15.5

Also enable sphinx for man generation.

nanoant committed:
LLVM and Clang 9.0.0 as independent packages

Ported from Arch

This is done in Arch way splitting everything into:
- llvm-libs
- llvm
- clang

Therefore clang will depend only on llvm-libs and does not pull whole llvm.

nanoant commented:
Okay, having `python3-sphinx` running was biggest PITA (hell lot of new deps), but it works. At least I am able to build `cmake` docs and LLVM build fine in my private branch (that one is not ready to merge yet).
Not I am reaving everywhere a reference to original package, e.g.:
~~~
# Ported from Arch Linux:
# https://git.archlinux.org/svntogit/packages.git/plain/trunk/PKGBUILD?h=packages/python-chardet
~~~
So it can be easily differ and updated.

nanoant commented:
Alright, I finally finished porting Clang 9.0.0 (see last commit in this PR). `/opt/arch/bin/clangd` now works with SublimeText on my Mac, see last commit in this PR.

<img width="493" alt="image" src="https://user-images.githubusercontent.com/103067/75112426-0d5a2d80-5644-11ea-95dd-2dc1349d32d7.png">

lloeki commented:
Cool, thanks.

I've done some serious work on my side to move to GitHub Actions on Catalina due to CircleCi pricing changes. Also basically rebuilt and updated many things locally. I'll do a PR but I need to produce a new base snapshot.

So I won't merge this in a little while til the above is complete.
